src/UI/SignTransactionsModals/SignWithDeviceModal/components/components/ConfirmAmount/ConfirmAmount.tsx

```diff
@@ -1,56 +1,116 @@
+import { useHandleAmountReference } from './hooks';
 
-export interface ConfirmAmountPropsType {
-  token: string;
-  formattedAmount: string;
-  rawAmount: string;
-  tokenAvatar?: string;
-  tokenType: TokenAvatarPropsType['type'];
-  tokenPrice?: number | null;
+export interface ConfirmAmountPropsType extends WithStylesImportType {
+  currentTransaction: ActiveLedgerTransactionType;
 }
 
 const ConfirmAmountComponent = ({
-  token,
-  tokenAvatar,
-  tokenType,
-  formattedAmount,
-  rawAmount,
-  tokenPrice,
-  styles
-}: ConfirmAmountPropsType & WithStylesImportType) => {
-  const isValidTokenPrice = tokenPrice != null;
-  const isLoadingTokenPrice = !isValidTokenPrice && tokenPrice !== null;
+  styles,
+  currentTransaction
+}: ConfirmAmountPropsType) => {
+  const { tokenId, nonce, amount } = currentTransaction.transactionTokenInfo;
+  const { isFontSizeLoading, handleAmountReference } =
+    useHandleAmountReference();
+
+  // If the token has a nonce means that this is an NFT. Eg: TokenId=TOKEN-1hfr, nonce=123 => NFT id=TOKEN-1hfr-123
+  const tokenIdForTokenDetails =
+    nonce && nonce.length > 0 ? `${tokenId}-${nonce}` : tokenId;
+
+  const { price: egldPrice } = useGetEgldPrice();
+  const { network } = useGetNetworkConfig();
+
+  const {
+    type,
+    name,
+    tokenAvatar,
+    tokenDecimals,
+    esdtPrice,
+    ticker,
+    identifier,
+    isLoading: isTokenDetailsLoading
+  } = useGetTokenDetails({
+    tokenId: tokenIdForTokenDetails
+  });
+
+  const isEgld = !tokenId;
+  const isNftOrSft = type
+    ? [NftEnumType.SemiFungibleESDT, NftEnumType.NonFungibleESDT].includes(type)
+    : false;
+
+  const tokenPrice = isEgld ? egldPrice : esdtPrice;
+  const egldLabel = getEgldLabel();
 
   return (
-    <div className={styles?.amount}>
-      <span className={styles?.label}>Amount</span>
+    <div className={styles?.confirmAmount}>
+      <div className={styles?.confirmAmountLabel}>
+        {isTokenDetailsLoading ? (
+          <LoadingDots />
+        ) : (
+          <ConfirmAmountLabel amount={amount} type={type} />
+        )}
+      </div>
 
-      <div className={styles?.token}>
-        <TokenAvatar type={tokenType} avatar={tokenAvatar} />
+      <div className={styles?.confirmAmountWrapper}>
+        <LoadingDots
+          className={classNames(styles?.confirmAmountLoading, {
+            [styles?.confirmAmountLoadingVisible]:
+              isTokenDetailsLoading || isFontSizeLoading
+          })}
+        />
 
         <div
-          className={styles?.value}
-          data-testid={DataTestIdsEnum.confirmAmount}
+          className={classNames(styles?.confirmAmountContent, {
+            [styles?.confirmAmountContentHidden]:
+              isTokenDetailsLoading || isFontSizeLoading
+          })}
         >
-          {formattedAmount} <TokenDetails.Label token={token} />
+          {isNftOrSft ? (
+            <ConfirmAmountNftSft
+              network={network}
+              name={name}
```

Comment:
why not pass the token as a whole object?

Reply:
Updated. 👍